shaders/src/light_indirect.fs

```diff
@@ -332,6 +332,91 @@ void evaluateSubsurfaceIBL(const PixelParams pixel, const vec3 diffuseIrradiance
 #endif
 }
 
+#if defined(HAS_REFRACTION)
+void applyRefraction(const PixelParams pixel,
+    const vec3 n0, const vec3 E, vec3 Fd, vec3 Fr,
+    inout vec3 color) {
+    // when reading from the cubemap, we are not pre-exposed so we apply iblLuminance
+    // which is not the case when we'll read from the screen-space buffer
+    float perceptualRoughness = pixel.perceptualRoughness;
+    float eta0 = pixel.etaOutIn;
+    vec3 r = -shading_view;
+
+    /* compute first interface refraction */
+#if REFRACTION_TYPE == REFRACTION_TYPE_SOLID
+    // for a sphere, with light at infinity (i.e. cubemap)
+    r = refract(r, n0, eta0);
+    float N0oR = dot(n0, r);
+#elif REFRACTION_TYPE == REFRACTION_TYPE_THIN
+    // Disney roughness remaping for thin layers
```

Comment:
Can you put the full citation ("Burley, 20xx, Title"). See brdf.fs for how I did it elsewhere.